orchestra/static/orchestra/task/js/required_fields.js

```diff
@@ -9,45 +9,73 @@
    */
   serviceModule.factory('requiredFields', function($rootScope, orchestraService) {
     var requiredFields = {
-      fields: [],
+      fields: {},
       invalid: [],
+      validators: {
+        'input-checkbox': [
+          function(elem) {
+            return elem.checked;
+          }
+        ],
+        'input-text': [
+          function(elem) {
+            return elem.value && elem.value.length > 0;
+          }
+        ],
+      },
       setup: function(data) {
         /**
          * Sets up the base data on which to validate fields.
          */
         this.data = data;
       },
-      require: function(fields) {
+      require: function(fieldType, field) {
         /**
-         * Sets a field as required. Fields are specified by
-         * dot-delimited key strings (e.g., `key0.key1.key2`).
+         * Sets a field as required. Fields are HTML elements to be
+         * checked by one or more validators according to their field
+         * type.
          */
-        this.fields = this.fields.concat(fields)
+        if (Object.keys(this.fields).indexOf(fieldType) >= 0) {
```

Comment:
is this the standard way to check for a key on an associative array? Doesn't just `if (this.fields[fieldType] != undefined)` work?